### DISCUSSION TITLE
Starlette is deprecating `Starlette` and `Router` decorators

### DISCUSSION THREAD

**Discussion by relsunkaev** [fastapi/fastapi#5593] [Questions] [answered]
[This pull request](https://github.com/encode/starlette/pull/1897) deprecates the use of decorators like `exception_handler` and `middleware`. Although it won't be removed until 1.0.0, I think a discussion on how to proceed would be useful.

**Comment by relsunkaev** [comment]
The PR is now part of Starlette 0.23

**Comment by Kludex** [comment, marked as answer]
Hmm... FastAPI will implement the deprecated methods by itself, to avoid users getting those warnings.
